presto-main/src/main/java/com/facebook/presto/execution/scheduler/filter/NodeSelectionCriteria.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.scheduler.filter;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public class NodeSelectionCriteria
```

Comment:
do we need a wrapper class around a collection of strings.
Is there a chance more fields will get added in the future?

Reply:
I would rather make this as the Predicate class since that is exactly what the name suggests instead of creating a Predicate wrapper later on.

Reply:
We expect more fields to be added into criteria for example source, user etc